src/go/rpk/pkg/vcloud/login.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package vcloud
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// audience of yak API in Auth0
+	// TODO(av) make it configurable to be able to work with production as well
+	yakAudience = "dev.vectorized.cloud"
+)
+
+var (
+	ErrTimeout = errors.New("authentication timed out")
+)
+
+type AuthToken struct {
+	Token string
+}
+
+// Login implements device flow of oauth2
+// reference https://auth0.com/docs/flows/call-your-api-using-the-device-authorization-flow
+func Login(auth0Client AuthClient) (*AuthToken, error) {
+	deviceCodeResponse, err := auth0Client.GetDeviceCode(yakAudience)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info(fmt.Sprintf("Go to the following link in your browser (code: %s):\n", deviceCodeResponse.UserCode))
+	log.Info(deviceCodeResponse.VerificationURIComplete)
+
+	log.Info("\nWaiting for authorization token to be issued...")
+	token, err := waitForToken(auth0Client, deviceCodeResponse)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("Success!")
+	log.Info(token.AccessToken)
```

Comment:
INFO - level logging should happen at the command layer, to prevent duplicate logs or spreading the logging over the code.

Reply:
I see the reasoning for that but the only way to achieve that would be to move all the logic to login instead of doing it here and I wanted to extract the whole login to package that could be tested. Thoughts?

Reply:
I'm not sure I understand, sorry. The `AccessToken` that you're logging is in `token`, so you can test that the value is present. Then, at the command (outer) layer, test that it's printed as expected.

In a nutshell, you can just move 

```
log.Info("Success!")
log.Info(token.AccessToken)
```
to the command

Reply:
oh sure, if that's enough, I'll do that. My note was about these logs https://github.com/vectorizedio/redpanda/pull/1263/files#diff-729e799cb03401524ddc0d2dabe7c3adefbe3cd0e9d565957eb31f95bcfe9a01R42 that I would like to stay in here